guides/auth-torii-with-github.md

```diff
@@ -0,0 +1,605 @@
+        'User-Agent': settings.USER_AGENT
+      }
+    };
+
+    const ghReq = https.request(options, (ghRes) => {
+      let body = '';
+      ghRes.setEncoding('utf8');
+      ghRes.on('data', (chunk) => body += chunk);
+      ghRes.on('end', () => {
+        res.writeHead(200, {
+          'Content-Type': 'application/json'
+        });
+        res.write(JSON.stringify(body));
+        res.end();
+      });
+    });
+    ghReq.on('error', (error) => {
+      console.error(error);
+      res.status(500).end();
+    });
+    ghReq.write(data);
+    ghReq.end();
+  });
+
+  app.use('/api/token', require('body-parser').json());
+  app.use('/api/token', tokenRouter);
+};
+```
+You may also want to add `"esversion": 6` to your `server/.jshintrc` to avoid warnings on this code.
+
+It creates a payload out of the `authorizationCode` and optionally `state` sent to it as well as the `client_id` and
+`client_secret` taken from your `settings.js`. It sends them to the `/login/oauth/access_token` end point,
+takes GitHub's response and returns it verbatim. The resulting access token will be in an `access_token` property of
+the JSON. We aren't going to run this through Ember Data, so the result does not need to be JSONAPI.
+
+Our application will need to know the URL of the token exchange API. While this is not an established part of the Torii
+configuration, that is a reasonable place to put it. Add `DEV_TOKEN_EXCHANGE_URL=http://localhost:4200/api/token` to your
+`.env` and the following to the development section of your `environment.js`. __Note that you should use `https` for the
+token exchange service in production!__
+
+```js
+// add to development section of config/environment.js
+
+ENV.torii.providers['github-oauth2'].tokenExchangeUri = process.env.DEV_TOKEN_EXCHANGE_URL;
+```
+
+Now that we've set up a token exchange service, let's use it. Replace your `torii` adapter with
+
+```js
+import Ember from 'ember';
+import ToriiAuthenticator from 'ember-simple-auth/authenticators/torii';
+import config from '../config/environment';
+
+export default ToriiAuthenticator.extend({
+  torii: Ember.inject.service(),
+  ajax: Ember.inject.service(),
+
+  authenticate() {
+    const ajax = this.get('ajax');
+    const tokenExchangeUri = config.torii.providers['github-oauth2'].tokenExchangeUri;
+
+    return this._super(...arguments).then((data) => {
+      return ajax.request(tokenExchangeUri, {
+        type: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify({
+          authorizationCode: data.authorizationCode
+        })
+      }).then( (response) => {
+        return {
+          access_token: JSON.parse(response).access_token,
+          provider: data.provider
+        };
+      });
+    });
+  }
+});
+```
+
+This makes a `POST` request to your service and gives the response back to Torii. Notice that it returns the `access_token`
```

Comment:
> gives the response back to Torii

That's not really what happens. Instead it gets the authorization code from torii (via `_super`) and exchanges that for an access token via the API service. torii is actually done and never called again when the promise returned from `_super` resolves.